.github/workflows/verify-staged-artifacts.yml

```diff
@@ -8,21 +8,42 @@ on:
         required: true
         type: string
 
+env:
+  ARTIFACTORY_USERNAME: ${{ secrets.ARTIFACTORY_USERNAME }}
+  ARTIFACTORY_PASSWORD: ${{ secrets.ARTIFACTORY_PASSWORD }}
+
 jobs:
-  verify-staged-with-jfrog:
+  verify-staged-with-sample:
     runs-on: ubuntu-latest
     steps:
 
-      - uses: jfrog/setup-jfrog-cli@v3
-        env:
-          JF_ENV_SPRING: ${{ secrets.JF_ARTIFACTORY_SPRING }}
+      - name: Checkout Sample Directory
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref_name }}
+          sparse-checkout: 'samples/time-spel-log'
+          sparse-checkout-cone-mode: false
+          show-progress: false
+
+      - name: Set up Gradle
+        uses: spring-io/spring-gradle-build-action@v2
 
-      - name: Download Artifact from Staging Repo
+      - name: Prepare Sample project against Staging
         run: |
-          fileToDownload=org/springframework/cloud/fn/spring-mail-supplier/${{ inputs.releaseVersion }}/spring-mail-supplier-${{ inputs.releaseVersion }}.jar
-          jfrog rt download libs-staging-local/$fileToDownload
-          if [ ! -f $fileToDownload ]
-          then
-            echo "::error title=No staged artifact::No spring-mail-supplier-${{ inputs.releaseVersion }}.jar in staging repository"
-            exit 1
-          fi
\ No newline at end of file
+          cd samples/time-spel-log
+          printf "allprojects {
+                  repositories {
+                    maven { 
```

Comment:
I don't think you will need a stronger `force` of the versions since these functions are not transitively brought in via Spring Boot.  For reference, we had to do something like [this](https://github.com/spring-projects/spring-pulsar/blob/1f2f6ed46d66a8cb7a146a1e4a742cfead6e0c2b/spring-pulsar-sample-apps/sample-apps-check-ci.gradle#L15-L22) for Spring Pulsar.

Reply:
No, we do not.
And you don't need to do that in Spring Pulsar as well.
There is just enough to override the version from Spring Boot: https://docs.spring.io/spring-boot/docs/current/gradle-plugin/reference/htmlsingle/#managing-dependencies.dependency-management-plugin.customizing

So, in your case that would be:
```
ext['spring-pulsar.version'] = "$springPulsarVersion"
```